includes/Core/Authentication/Authentication.php

```diff
@@ -991,4 +1005,93 @@ private function redirect_to_proxy( $code ) {
 		);
 		exit;
 	}
+
+	/**
+	 * Sets the current site URL if the current user has setup capabilities.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function set_connected_proxy_url() {
+		$this->connected_proxy_url->set( home_url() );
+	}
+
+	/**
+	 * Checks whether the current site URL has changed or not. If the URL has been changed,
+	 * it disconnects the Site Kit and sets the disconnected reason to "connected_url_mismatch".
+	 *
+	 * @since n.e.x.t
+	 */
+	private function check_connected_proxy_url() {
+		if ( $this->connected_proxy_url->matches_url( home_url() ) ) {
+			return;
+		}
+
+		if ( ! current_user_can( Permissions::SETUP ) ) {
+			return;
+		}
+
+		if ( ! $this->credentials->has() ) {
+			return;
+		}
+
+		if ( ! $this->credentials->using_proxy() ) {
+			return;
+		}
+
+		if ( ! $this->is_authenticated() ) {
+			return;
+		}
+
+		if ( ! $this->connected_proxy_url->has() ) {
+			$this->set_connected_proxy_url();
+			return;
+		}
+
+		$this->disconnect();
+		$this->disconnected_reason->set( Disconnected_Reason::REASON_CONNECTED_URL_MISMATCH );
+	}
+
+	/**
+	 * Handles user connection action and redirects to the proxy connection page.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function handle_sync_site_fields() {
+		$googlesitekit_code = $this->context->input()->filter( INPUT_GET, 'googlesitekit_code' );
+		if ( $googlesitekit_code ) {
+			return;
+		}
+
+		if ( ! current_user_can( Permissions::SETUP ) ) {
+			wp_die( esc_html__( 'You have insufficient permissions to connect Site Kit.', 'google-site-kit' ) );
+		}
+
+		if ( ! $this->credentials->using_proxy() ) {
+			wp_die( esc_html__( 'Site Kit is not configured to use the authentication proxy.', 'google-site-kit' ) );
+		}
+
+		if ( $this->disconnected_reason->get() === Disconnected_Reason::REASON_CONNECTED_URL_MISMATCH ) {
+			$this->google_proxy->sync_site_fields( $this->credentials, 'sync' );
+		}
+
+		$this->redirect_to_proxy( '' );
```

Comment:
Technically, this isn't necessary here, since the hook at priority 10 for the same action will already do that redirect.

Reply:
Ok, removed.